src/vos/vos_dtx.c

```diff
@@ -2036,14 +2092,18 @@ vos_dtx_commit(daos_handle_t coh, struct dtx_id *dtis, int count,
 	rc = umem_tx_begin(vos_cont2umm(cont), NULL);
 	if (rc == 0) {
 		committed = vos_dtx_commit_internal(cont, dtis, count,
-						    0, dcks, daes);
+						    0, dcks, daes, dces);
 		rc = umem_tx_end(vos_cont2umm(cont),
 				 committed > 0 ? 0 : committed);
 		if (rc == 0)
-			vos_dtx_post_handle(cont, daes, count, false);
+			vos_dtx_post_handle(cont, daes, NULL, count, false);
+		else
```

Comment:
could potentially add a local to store either NULL or dces if rc != 0

Reply:
We can free "dces" before calling vos_dtx_post_handle() for "rc == 0" case, then always use "dces".